tests/migrations/test_operations.py

```diff
@@ -5182,6 +5182,319 @@ def assertModelsAndTables(after_db):
+                fields=["-author", "title"],
+                name="author_title_index",
+            )
+            assert_index_not_created(
+                [
+                    migrations.AddIndex("Book", negative_index),
+                    migrations.RemoveIndex("Book", negative_index.name),
+                ]
+            )
+
+    def test_composed_indexes_dont_create_dup_implicit_fk_index_when_deleting(self):
+        """
+        When we delete a composed index with a FK included, we shouldn't create the
+        implicit index, when it is included in another index
+        """
+
+        app_label = "test_cidcdfkid"
+
+        create_author = migrations.CreateModel(
+            "Author",
+            [("id", models.BigAutoField(primary_key=True))],
+            options={},
+        )
+        create_book = migrations.CreateModel(
+            "Book",
+            [
+                ("id", models.BigAutoField(primary_key=True)),
+                ("author", models.ForeignKey(to="Author", on_delete=models.CASCADE)),
+                ("title", models.CharField(max_length=100)),
+                ("pub_date", models.DateField()),
+            ],
+            options={},
+        )
+
+        project_state = self.apply_operations(
+            app_label,
+            ProjectState(),
+            [
+                create_author,
+                create_book,
+            ],
+        )
+
+        def get_author_field_indexes():
+            Book = project_state.apps.get_model(app_label, "Book")
+            columns = ["author_id"]
+            with connection.schema_editor() as editor:
+                return editor._constraint_names(Book, index=True, column_names=columns)
+
+        def assert_index_not_created(operations):
+            initial_indexes = get_author_field_indexes()
+            self.apply_operations(app_label, project_state, operations)
+            actual_indexes = get_author_field_indexes()
+            self.assertEqual(set(actual_indexes), set(initial_indexes))
+
+        unique_constraint = models.UniqueConstraint(
+            fields=["author", "title"],
+            name="author_title_uniq",
+        )
+        simple_index = models.Index(
+            fields=["author", "title"],
+            name="author_title_index",
+        )
+        f_index = models.Index(
+            models.F("author"),
+            models.F("title"),
+            name="author_title_index",
+        )
+        # combinations of operations that would create an additional index
+        # here we shouldn't do that as another index exists and includes the FK field
+        operations_to_check = [
+            [
+                migrations.AlterUniqueTogether("Book", ["author", "title"]),
+                migrations.AlterUniqueTogether("Book", []),
+            ],
+            [
+                migrations.AlterIndexTogether("Book", ["author", "title"]),
+                migrations.AlterIndexTogether("Book", []),
+            ],
```

Comment:
Twice the same.
Did you mean:
```suggestion
            [
                migrations.AlterUniqueTogether("Book", ["author", "title"]),
                migrations.AlterUniqueTogether("Book", []),
            ],
            [
                migrations.AlterIndexTogether("Book", []),
                migrations.AlterIndexTogether("Book", ["author", "title"]),
            ],
```
?